src/include/small_encoding.h

```diff
@@ -0,0 +1,39 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+#ifndef CEPH_SMALL_ENCODING_H
+#define CEPH_SMALL_ENCODING_H
+
+#include "include/buffer.h"
+#include "include/int_types.h"
+
+// first(low) 2 bits = how many low nibbles are zeros
+// high bit of each byte = another byte follows
+// (so, 5 bits data in first byte, 7 bits data thereafter)
```

Comment:
Could also make 00 = no low zeros, 01 = 6 low zeros, 02 = 10 low zeros, 03 = 14 low zeros, or something like that if we think the <6 or even <12 low zero bits is less common to optimize for.  Remember block_size is 4096 = 12 bits.